### ISSUE TITLE
Same series points lead to different extremes on y-Axis if chart height is smaller than 250px

### ISSUE THREAD

**Issue by sergevic** [highcharts/highcharts#6811]
#### Expected behaviour
Min and max values for extremes remains the same if input data is the same. It should also be independent of chart container height.

#### Actual behaviour
Actual behaviour depends on the chart height. 
If the chart height is less that 250px y-Axis range changes after same data assigned via series.setData() method.
If the chart height 250px y-Axis range remains the same as expected.
<img src="https://monosnap.com/file/mJ6Tkg0lWsMYgqClF0VLJylSMk37JR.png">

#### Live demo with steps to reproduce
http://jsfiddle.net/0c4k66qt/1/

#### Affected browser(s)
all browsers (tested in Chrome/IE/Firefox).

#### Workaround
Completely recreate the series.

**Comment by pawelfus** [comment]
Hi @sergevic 

Thank you for this report. Could you set fixed width for the chart? For me it works fine.

To explain: this may happen, because of default value for [`yAxis.tickPixelInterval`](http://api.highcharts.com/highcharts/yAxis.tickPixelInterval). It's an approximate value, so tick positions may vary. I know the demo sets the same dataset, but after `setData` we have all necessary data (xAxis&yAxis labels, title, subtitle, legend etc) to calculate precise plotting area width and height.

**Comment by sergevic** [comment]
Hi @pawelfus 
I'm not sure how "set fixed width" can help. I tried to set width both via css and chart: { width: ... } configuration, but the issue is still here http://jsfiddle.net/0c4k66qt/2/
I also did not catch explanation: the chart height is predefined/fixed, so there is no any guessing about height dimensions. So process of calculation yAxis range and ticks, from my point of view, should be deterministic: e.g. produce same results both during initialization and setData.

Problem can be solved if applied value <= 70 to yAxis.tickPixelInterval. But if apply value, say 71 (default is 72) the problem is still here. 
So I would not say this is a solution, but definitely is the workaround.

**Comment by pawelfus** [comment]
I was hoping it will help to recreate issue on my side with a fixed width, as I noticed xAxis labels are rendered in a differently. It does not apparently: it works fine on OSX (chrome), but I was able to recreate issue on Windows (chrome too). That most probably means browsers on Windows returns different labels heights.

The height is predefined but not the plotting area height (sorry for not being clear, by plotting area I mean area where data/lines/bars are rendered, not the chart). At the same time we need to reserve space for xAxis labels and yAxis labels (and Axes titles, which we may need to wrap). For example we can first reserve space for xAxis, labels fit horizontally without rotation, but after calculating yAxis space, we need to rotate xAxis labels. Then again, we need to check yAxis if we need to apply other format for labels (e.g. ellipsis) etc. etc. etc. That is the reason why `tickPixelInterval` applies approximately distance between ticks. 

Let me know if something is not clear.
